[1] - Redesigned routes of Trainee role

akash-kumar-dev opened the pull request:
Addresses: https://github.com/a2i-code-For-Govstack/nise3-skill-and-certification/issues/1

https://github.com/user-attachments/assets/913edd61-8951-4e00-90a1-cfa05724a187

## redesigned Trainee Pages
- Trainee-profile
- edit personal info
- add Education
- add guardian Info

{routes: /trainee-profile, /edit-personal-info, /add-edit-education/2, /add-guardian-info )

Shelly011s commented:
{routes: , /edit-personal-info, /add-edit-education/2, /add-guardian-info )
- /trainee-profile : Remove the + icon and try to center the text in button.
- In all the routes fix the back to profile button as discussed in figma design and in the header of each section center the text.
- Change the profile dropdown on hover color to something dark purple.(Currently its white making it unable to the read the text)
- /add-edit-education : Remove the card inside the card where the inputs are and the text after button "Like (CSE) or (Science)" should be "*Like (CSE) or (Science)" and also add space around.
Please go through comments added in code

Shelly011s commented on resources/views/frontend/trainee/trainee-profile.blade.php:
```diff
@@ -5,51 +5,84 @@
+                            <h4>Personal Information</h4>
                         </div>
 
                         <div class="card-tools">
```
Try doing this for all the buttons to center the text. 
 <div class="card-tools btn btn-primary d-flex align-items-center justify-content-center btn-rounded custom-button-style px-4 py-2 font-weight-bold">
                            <a href="{{ route('frontend.edit-personal-info') }}"
                                class="">
                                {{ __('admin.common.edit') }}
                            </a>
                        </div>

Shelly011s replied:
Try doing this for all the buttons to center the text. 
```
 <div class="card-tools btn btn-primary d-flex align-items-center justify-content-center btn-rounded custom-button-style px-4 py-2 font-weight-bold">
                            <a href="{{ route('frontend.edit-personal-info') }}"
                                class="">
                                {{ __('admin.common.edit') }}
                            </a>
                        </div>
```

Shelly011s commented on resources/views/frontend/trainee/trainee-profile.blade.php:
```diff
@@ -5,51 +5,84 @@
                     <div class="card-body">
-                        <div class="row">
-                            <div class="col-md-2">
+                        <div class="row justify-content-center">
```
```
 <div class="row justify-content-flex-start">
                            <div class="col-md-4 col-lg-3 col-xl-2 mr-xl-5 mr-lg-4 ml-lg-4 d-flex justify-content-center">
```

akash-kumar-dev commented:
fixed issues:
- removed + icon or other icons from buttons
- changed back to profile button as in Figma
- fixed profile dropdown

form previous PR fixed issues https://github.com/a2i-code-For-Govstack/nise3-skill-and-certification/pull/5

- replaced card-header border with `hr` tag
- fixed card size

Shelly011s commented on resources/views/frontend/trainee-profile/add-guardian-information.blade.php:
```diff
@@ -7,13 +7,53 @@
+            font-weight: normal !important;
+        }
+
+        .back-to-profile {
```
`.back-to-profile {
            background-color: #ffffff;
            border-radius: 12px;
            box-shadow: 0px 4px 4px #00000040;
            color: #671688;
            font-weight: 700;
        }
` Change every back to profile button to this

Shelly011s commented on resources/views/frontend/trainee-profile/add-edit-education-one.blade.php:
```diff
@@ -1,262 +1,339 @@
+                    </div>
+
+
+                    <div class="row w-100 d-flex justify-content-center">
```
```
<div class="row w-100 d-flex justify-content-around">
                      
                            <button class="btn btn-info btn-rounded px-5 pb-1 pt-2 add_form_field" 
                                style="border-radius: 12px;">{{ __('frontend.trainee.add_more') }} <i class="fa fa-plus"
                                    aria-hidden="true"></i></button>
                        
                        
                            <button type="submit" class="btn btn-success btn-rounded px-5 pb-1 pt-2 custom-button-style"
                                >{{ __('generic.save') }}</button>
                        
                    </div>
```
#### Change this entire block for save and add more button.

Shelly011s commented on resources/views/frontend/trainee-profile/add-edit-education-one.blade.php:
```diff
@@ -1,262 +1,339 @@
+            font-weight: normal !important;
+        }
+
+        .back-to-profile {
```
```
.back-to-profile {
            background-color: #ffffff;
            border-radius: 12px;
            box-shadow: 0px 4px 4px #00000040;
            color: #671688;
            font-weight: 700;
        }
``` 
Change every back to profile button to this

Shelly011s commented on resources/views/frontend/trainee-profile/add-edit-education-one.blade.php:
```diff
@@ -1,262 +1,339 @@
-                                       class="btn btn-sm btn-outline-primary btn-rounded">
-                                        <i class="fas fa-backward"></i> {{ __('generic.back_to_profile') }}
+                                    <a href="{{ route('frontend.trainee') }}"
+                                        class="btn btn-sm btn-rounded back-to-profile font-weight-bold">
```
```
<a href="{{ route('frontend.trainee') }}"
                                        class="btn btn-sm btn-rounded back-to-profile font-weight-bold px-4 pt-2 pb-1">
```
Add this change as well across the back to profile button

Shelly011s commented on resources/views/frontend/trainee/trainee-profile.blade.php:
```diff
@@ -5,51 +5,86 @@
+                            <h4 style="transform: translateY(30%)">Personal Information</h4>
                         </div>
 
                         <div class="card-tools">
```
```
<a href="{{ route('frontend.edit-personal-info') }}"
                                class="card-tools btn btn-sm btn-primary custom-button-style btn-rounded px-4 pt-2 pb-1 font-weight-bold">
                                <!-- <i class="fas fa-plus-circle"></i> -->
                                {{ __('admin.common.edit') }}
                            </a>
                       </div>
                    <div class="card-body">
                        <div class="row justify-content-flex-start">
                            <div class="col-md-4 col-lg-3 col-xl-2 mr-xl-5 mr-lg-4 ml-lg-4 d-flex justify-content-center">
```
`Remove <div class="card-tools">` and add the above piece of code.
Next after ` <div class="card-body">`
```
<div class="row justify-content-flex-start">
                            <div class="col-md-4 col-lg-3 col-xl-2 mr-xl-5 mr-lg-4 ml-lg-4 d-flex justify-content-center">
```

Shelly011s commented on resources/views/frontend/trainee-profile/edit-personal-info.blade.php:
```diff
@@ -20,174 +56,174 @@
+                                                        <label for="profile_pic">
+                                                            <img class="img-thumbnail rounded-circle"
+                                                                src="{{ $trainee->profile_pic ? asset('storage/' . $trainee->profile_pic) : 'https://via.placeholder.com/350x350?text=Profile+Picture' }}"
+                                                                style="width: 300px; height: 300px" alt="Profile pic" />
```
```
style="width: 240px; height: 200px" alt="Profile pic" />
                                                            <span class="py-1 px-2 bg-gray"
                                                                style="position: absolute; right: 50%; bottom: 0; border: 2px solid #afafaf; border-radius: 50%; overflow: hidden">
```
Change this too

Shelly011s commented on resources/views/frontend/trainee-profile/edit-personal-info.blade.php:
```diff
@@ -20,174 +56,174 @@
-                                    >
-                                </div>
+                                    <div class="col-md-12">
+                                        <input type="submit"
```
```
<input type="submit"
                                            class="btn btn-sm btn-primary float-right px-3 pt-2 pb-1 custom-button-style"
                                            value="{{-- {{ __('admin.common.edit') }} --}} Save Changes">
```
Just copy this code. I have fixed the spelling as well as stying issues for alignment that you missed.

Shelly011s commented on resources/views/frontend/trainee-profile/edit-personal-info.blade.php:
```diff
@@ -20,174 +56,174 @@
-                                    <input type="submit" class="btn btn-primary float-right ml-2"
-                                           value="{{ __('admin.common.edit') }}">
+                                        <div class="col-md-12">
+                                            <input type="submit"
```
`<input type="submit"
                                                class="btn btn-sm btn-primary float-right custom-button-style px-3 pt-2 pb-1"
                                                value="{{-- {{ __('admin.common.edit') }}  --}}Change Password">` Copy this code as well to fix styling issue

Shelly011s commented on resources/views/frontend/search-courses/course-list.blade.php:
```diff
@@ -18,9 +18,12 @@
         <div class="row">
             <div class="col-md-12">
```
```
<div class="col-md-12 p-0">
                <div class="card mb-0">
```
Similarly, just copy the given for `index.blade.php` and `skill-videos.blade.php`.

Shelly011s commented on resources/views/frontend/search-courses/course-list.blade.php:
```diff
@@ -18,9 +18,12 @@
+                            {{__('generic.courses')}}
+                        </h2>
                     </div>
+                    <hr class="mt-2 mb-0" style="background-color: transparent; border: 1px solid rgba(0, 0, 0, 0.125);">
```
` <hr class="mt-2 mb-0 mx-4" style="border: 0; height: 1px; background-image: linear-gradient(to right, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0));">`
Similarly, just copy the given for all the `<hr>` tags in `index.blade.php` and `skill-videos.blade.php`.

akash-kumar-dev commented:
All issues have been resolved. I have added and fixed all the issues that were raised.
